core/whistle-1.0.0/src/wh_json.erl

```diff
@@ -334,7 +334,7 @@ all(Pred, ?JSON_WRAPPER(Prop)) when is_function(Pred, 1) ->
 any(Pred, ?JSON_WRAPPER(Prop)) when is_function(Pred, 1) ->
     lists:any(Pred, Prop).
 
--spec foldl(fun((key(), json_term(), any()) -> any()), any(), object()) -> any().
+-spec foldl(fun((key(), json_term(), any()) -> Ret), Ret, object()) -> Ret.
```

Comment:
I don't think this is right. Dialyzer would assume all must be the same value, but since this is the accumulator param, we can't know what the caller is passing in nor how the function F will manipulate said accumulator.